2324-wt/abstract-interpretation/interim/poster.tex

```diff
@@ -0,0 +1,221 @@
+
+		\begin{scope}[no,transparency group]
+			\node[n,roundnode,below=of Y2] (Y3) {\rc{y}\textsubscript4};
+			\node[n,squarenode,below=of C1] (C2) {\rc{compB}};
+			\node[ds] at (Y3.north east) {\tiny\tbot};
+		\end{scope}
+
+		\node[n,roundnode,above=of Y1] (Y4) {\rc{y}\textsubscript2};
+
+		\node[dr] at (X1.north east) {\tiny$\{10\}$};
+		\node[ds] at (X2.north east) {\tiny$\{10\}$};
+		\node[dr] at (Y1.north east) {\tiny$\{15\}$};
+		\node[ds] at (Y2.north east) {\tiny$\{15\}$};
+		\node[ds] at (Y4.north east) {\tiny$\{15\}$};
+
+		\node[t,fit=(X1)(X2)(Y1)(Y2)(Y3)(Y4)(C1)(C2)] (F) {};
+		% \node[label] at (F.south) {Step 3};
+
+		\draw[a] (X1) to (X2);
+		\draw[a] (X2) to (Y1);
+		\draw[a] (Y1) to (Y2);
+		\draw[a] (Y2) to (C1);
+
+		\begin{scope}[no,transparency group]
+			\draw[a] (Y1) to (Y3);
+			\draw[a] (Y3) to (C2);
+		\end{scope}
+
+		\draw[a] (Y1) to (Y4);
+	\end{tikzpicture}\par
+	% \paragraph{Step 3} The last step uses the decorated dataflow graph to determine wich
+	% path can never be executed. The identified path will then be discarded from the
+	% graph.\par
+	\paragraph{Step 3} The last step uses the decorated dataflow graph to determine and
+	track different properties. We can, e.g., catch endless loops, or, as shown above,
+	eliminate dead paths in the dataflow graph.\par
+\end{multicols}\bigskip
+\endgroup
+
+\begin{multicols}{3}
+	\begin{minipage}{\dimexpr2\columnwidth+\columnsep\relax}
+		\section*{How abstract interpretation fits into \flowr}
+		\paragraph{\flowr}\flowr{} is a static \emph{dataflow analyzer} and
+		\emph{program slicer} for the R language. Program slicers reduce a given program
+		to a set of statements --- the so called slices --- that influence a variable at
+		a specific point in the program --- the slicing criterion. \flowr{} formulates
+		the slicing as a reachability problem on the dataflow graph, which itself is
+		based on the abstract syntax tree of the given R~program.
+		\paragraph{Abstract Interpretation} Abstract interpretation is all about
+		figuring out \emph{runtime properties} of a given program without actually
+		executing it. These properties include aspects like the sign or nullness of a
+		variable. For now we decided to limit ourselves to determining the \emph{domain
+		of numeric variables}. In other words, we want to know what values a variable
+		can have at a certain point in the program.
+		\paragraph{Abstract interpretation in \flowr} Abstract interpretation helps
+		\flowr{} to further \emph{narrow down the slice} of a program. This can be done
+		by removing certain paths from the dataflow graph if we can prove that they will
+		never be executed. Imagine a conditional like \rc{if(i < 10) doit() else run()}.
+		If, through abstract interpretation, we know that \rc{i} is always smaller than
+		\rc{10}, there's no need to include the path stemming from the \rc{if}'s
+		else-case in the slice. % TODO: is this okay? Or does it read like we only catch dead code
```

Comment:
ich finde das passt, wenn du magst kannst du aber ein `e.g.` einweben.

Reply:
Ich habs jetzt 20 mal gelesen und bin jetzt gar nicht mehr zufrieden damit xD

Was hältst du von
>  ... This can be done by removing nodes from the dataflow graph that are contained in a provably dead path in the controlflow graph.